sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                    'https://skypilot.readthedocs.io/en/latest/getting-started/installation.html.' # pylint: disable=line-too-long
+                    )
+
+    @classmethod
+    def validate_name(cls, name) -> str:
+        """Validates the name of the AZ Container.
+
+        Source for rules: https://learn.microsoft.com/en-us/rest/api/storageservices/Naming-and-Referencing-Containers--Blobs--and-Metadata#container-names # pylint: disable=line-too-long
+
+        Args:
+            name: str; name of the container
+
+        Returns:
+            name: str; name of the container
+
+        Raises:
+            StorageNameError: if the given container name does not follow the
+                naming convention
+        """
+
+        def _raise_no_traceback_name_error(err_str):
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.StorageNameError(err_str)
+
+        if name is not None and isinstance(name, str):
+            if not 3 <= len(name) <= 63:
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} must be between 3 (min) '
+                    'and 63 (max) characters long.')
+
+            # Check for valid characters and start/end with a letter or number
+            pattern = r'^[a-z0-9][-a-z0-9]*[a-z0-9]$'
+            if not re.match(pattern, name):
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} can consist only of '
+                    'lowercase letters, numbers, and hyphens (-). '
+                    'It must begin and end with a letter or number.')
+
+            # Check for two adjacent hyphens
+            if '--' in name:
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} must not contain '
+                    'two adjacent hyphens.')
+
+        else:
+            _raise_no_traceback_name_error('Store name must be specified.')
+        return name
+
+    def initialize(self):
+        """Initializes the AZ Container object on the cloud.
+
+        Initialization involves fetching bucket if exists, or creating it if
+        it does not. Also, it checks for the existance of the storage account
+        and resource group if provided by the user. If not provided, those are
+        created with a default naming convention.
+
+        Raises:
+            StorageBucketCreateError: If container creation fails or storage
+                account attempted to be created already exists.
+            StorageBucketGetError: If fetching existing bucket fails.
+            StorageInitError: If general initialization fails.
+        """
+        self.storage_client = data_utils.create_az_client('storage')
+        self.resource_client = data_utils.create_az_client('resource')
+        # Using externally created storage
+        if isinstance(self.source, str
+                      ) and data_utils.is_az_container_endpoint(self.source):
```

Comment:
```suggestion
        if (isinstance(self.source, str
                      ) and data_utils.is_az_container_endpoint(self.source)):
```
and re-run formatter

Reply:
Done